### PR TITLE
feat(useElementCross): new function

### PR DESCRIPTION
**Pull request by JiatLn**
### Description

Reactive two element whether cross state.

### Additional context

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [X] New Feature
- [X] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [X] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [X] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [X] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useElementCross/index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { computed } from 'vue-demi'
+import type { MaybeComputedElementRef } from '../unrefElement'
```
Comment (songjianet) [inline packages/core/useElementCross/index.ts:2]: import type is best placed at the bottom of all references.

**Review by songjianet — APPROVED** [review]
Can be used as a reference.

**Comment by chris-zhu** [comment]
I think we should improve the demo. Like [`useDraggable`](https://github.com/vueuse/vueuse/blob/main/packages/core/useDraggable/demo.vue). Move those two elements to observe more values of `isCross`

**Comment by wheatjs** [comment]
A couple of thoughts:

1. This seems a little specific, I think we would probably want something more generic that would accept more elements.
2. If we did add this function, I think something like `useElementIntersection` might be a better name

Tbh though, I feel like this type of function is more of an edge case and could be left up to the user. Otherwise we might provide something that isn't flexible enough for the cases where it might be used and not common enough for general use.

**Comment by darrylnoakes** [comment]
How does this relate to #1638?

**Pull request closed without merging**